spec/requests/api/v1/addresses/index_spec.rb

```diff
@@ -0,0 +1,59 @@
+describe 'GET /api/v1/user/addresses', type: :request do
+  subject { get api_v1_user_addresses_path, headers: headers, as: :json }
+
+  let(:headers) { auth_headers }
+  let(:user) { create(:user) }
+  let(:city) { create(:city) }
+
+  context 'when there are addresses created' do
+    let(:addresses) { create_list(:address, 2, user: user, city: city) }
+
+    before do
+      addresses
```

Comment:
```suggestion
    let!(:addresses) { create_list(:address, 2, user: user, city: city) }

    before do
```

Reply:
in case you want to know how it works

https://relishapp.com/rspec/rspec-core/v/2-11/docs/helper-methods/let-and-let